src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -255,6 +243,19 @@ internal TabItem AddTabItem(IViewExtension viewExtension, ContentControl content
             return null;
         }
 
+        /// <summary>
+        /// This method will close a tab item in the right side bar based on passed extension
+        /// </summary>
+        /// <param name="viewExtension">Extension to be closed</param>
+        /// <returns></returns>
+        internal void CloseTabItem(IViewExtension viewExtension)
+        {
+            string tabName = viewExtension.Name;
+            TabItem tabitem = TabItems.OfType<TabItem>().SingleOrDefault(n => n.Header.ToString() == tabName);
+            CloseExtension?.Invoke(tabName);
```

Comment:
closeTab (below) is already raising this event... why do you need to raise it twice?

Reply:
CloseTab(object sender, RoutedEventArgs e) is triggered when the close button is clicked by the user and this event is raised to handle the Workspace references menu item.

Reply:
ahh, would you mind renaming the eventHandler so the name does not conflict, would clear up some confusion here I think?

Reply:
`CloseTabEventHandler`? maybe?

Reply:
Resolving this as it is addressed by Aaron's PR